web/packages/teleport/src/Bots/Details/JoinMethodIcon.tsx

```diff
@@ -117,23 +111,8 @@ const renderIcon = (
   }
 };
 
-const ResourceIconContainer = styled(Flex)<{ size: IconProps['size'] }>`
-  width: ${({ size }) => sizetoOuterPx(size)};
-  height: ${({ size }) => sizetoOuterPx(size)};
-  align-items: center;
-  justify-content: center;
-`;
-
-function sizetoOuterPx(size: IconProps['size']) {
-  if (size === 'small') return '16px';
-  if (size === 'medium') return '20px';
-  if (size === 'large') return '24px';
-  if (size === 'extra-large') return '32px';
-  return '24px';
-}
-
-function sizetoInnerPx(size: IconProps['size']) {
-  if (size === 'small') return '12px';
+function sizetoPx(size: IconProps['size']) {
```

Comment:
This function mirrors a part of the logic that belongs to the inside of the `Icon` component, so consider extracting it to a separate common function. Secondly, since we started using `IconProps` to define sizes of `ResourceIcon`, it may be a better idea to port the concept of an icon size enum to the `ResourceIcon` component itself and handle it there, with `IconProps['size'] extracted as a separate type, perhaps common to both `Icon` and `ResourceIcon`.